pkg/apiserver/resthandler.go

```diff
@@ -171,6 +172,85 @@ func CreateResource(r RESTCreater, ctxFn ContextFunc, namer ScopeNamer, codec ru
 	}
 }
 
+// PatchResource returns a function that will handle a resource patch
+func PatchResource(r RESTPatcher, ctxFn ContextFunc, namer ScopeNamer, codec runtime.Codec, resource string, admit admission.Interface) restful.RouteFunction {
+	return func(req *restful.Request, res *restful.Response) {
+		w := res.ResponseWriter
+
+		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
+		timeout := parseTimeout(req.Request.URL.Query().Get("timeout"))
+
+		namespace, name, err := namer.Name(req)
+		if err != nil {
+			notFound(w, req.Request)
+			return
+		}
+		ctx := ctxFn(req)
+		ctx = api.WithNamespace(ctx, namespace)
+
+		original, err := r.Get(ctx, name)
```

Comment:
My next set of changes to update will make implementing patch under AtomicUpdate feasible.  For now, that doesn't need to block this change.  We should leave a todo here to that effect.


> On Mar 1, 2015, at 2:08 PM, Mike Danese <notifications@github.com> wrote:
> 
> In pkg/apiserver/resthandler.go:
> 
> > +func PatchResource(r RESTPatcher, ctxFn ContextFunc, namer ScopeNamer, codec runtime.Codec, resource string, admit admission.Interface) restful.RouteFunction {
> > +	return func(req *restful.Request, res *restful.Response) {
> > +		w := res.ResponseWriter
> > +
> > +		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
> > +		timeout := parseTimeout(req.Request.URL.Query().Get("timeout"))
> > +
> > +		namespace, name, err := namer.Name(req)
> > +		if err != nil {
> > +			notFound(w, req.Request)
> > +			return
> > +		}
> > +		ctx := ctxFn(req)
> > +		ctx = api.WithNamespace(ctx, namespace)
> > +
> > +		original, err := r.Get(ctx, name)
> @lavalamp The only issue with that is only a few Update implementations do reads before update. RESTs that use AtomicUpdate do read and most that use SetObj do not read. It might be nice to standardize on an Update mechanism that could skip the initial read when unnecessary.
> 
> —
> Reply to this email directly or view it on GitHub.
>